### DISCUSSION TITLE
cannot deploy shiny apps to shiny.io

### DISCUSSION THREAD

**Discussion by olgadiella** [teamalgoritma/community#112] [Q&A] [answered]
Hi Team,

Saya coba deploy shiny apps ke shiny.io namun gagal dengan informasi sbb:
"An error has occurred
The application failed to start. Contact the author for more information."

Shiny.io sdh connect ke R Studio, sdh copas secret, data yg digunakan jg sdh include saat deploy.

What to do?
pls advise, thank you

**Comment by inytss** [comment]
Dear @olgadiella ,
Untuk mengetahui informasi error yang terjadi, silahkan untuk melakukan pengecekan pada log shinyapps yang dideploy pada akun shinyapps.io Anda. Boleh dicek pada bagian log app terbawah jika terdapat pesan error, maka Anda perlu memperbaiki error yang terjadi tersebut.

**Reply by olgadiella** [reply]
Dear Team,

Berikut ss log nya, pls help, thank you
![WhatsApp Image 2022-05-17 at 12 17 59 PM](https://user-images.githubusercontent.com/103428684/168734683-88cc0327-87db-421c-857b-50e23aabfa29.jpeg)
![WhatsApp Image 2022-05-17 at 12 18 10 PM](https://user-images.githubusercontent.com/103428684/168734688-50c7096d-741f-439b-a782-7f2d28e97cdd.jpeg)
![WhatsApp Image 2022-05-17 at 12 18 34 PM](https://user-images.githubusercontent.com/103428684/168734695-64264633-1161-41c5-a259-a99b118951be.jpeg)

Berikut coba run library tidyverse, artinya apa ya?
![image](https://user-images.githubusercontent.com/103428684/168735545-d2bf726d-9e4a-43fd-a49e-0243a7ee3dae.png)

**Reply by inytss** [reply]
Boleh dicoba untuk membersihkan global environment lalu dicoba kembali untuk menjalankan shinyappsnya di local apakah dapat berjalan dengan baik?

**Reply by olgadiella** [reply]
mb @inytss sdh coba dg ctrl+shift+f10, kmdn di-deploy, namun masih gagal, pls advise, trm ksh

![image](https://user-images.githubusercontent.com/103428684/168789141-116a47f4-cf5a-4c62-bf14-34f8263f860c.png)

**Reply by aitanarP-Heryuda** [reply]
Dear Mba Olga,
kalau error nya sama mungkin bisa cek disini : 
https://community.rstudio.com/t/unable-to-deploy-shiny-app-in-shinyapps-io-the-application-failed-to-start-exited-with-code-137/18787

atau coba ini

 options(shiny.maxRequestSize=200*1024^2),

**Reply by olgadiella** [reply]
Dear Pak Heryuda, 
Trm ksh ya Pak, sudah coba cek namun sepertinya beda case Pak :')

**Reply by aitanarP-Heryuda** [reply]
library tidyverse nya udah coba dihapus mba? iseng nyobain punyaku tetep bisa jalan ,
mungkin worth to try.. tapi ya gatau juga apa bakal jalan.. :D

**Reply by olgadiella** [reply]
Kmrn saya coba create baru lg dr awal utk ui, server, dan globalnya, Pak @aitanarP-Heryuda dan mb @inytss, 
kmdn coba publish, namun msh failed jg, pas cek log, skrg yg error plotly. Coba searching info dan ketemu case serupa, ngikutin caranya: unpackage plotly, kmdn install ulang, namun msh failed jg :')

Sampai kmrn malam ngubek2 info dr bbrp community dg case serupa dan coba apply cara mrk tp msh blm berhasil jg, sedih bgt pdhl uda create codingnya tp blm brhasil deploy

![image](https://user-images.githubusercontent.com/103428684/168936590-bed362df-a5ae-41d8-8acf-1600907c0e6a.png)
![image](https://user-images.githubusercontent.com/103428684/168936616-4c65cd8f-64a1-4892-9c8e-f42d8d680d6e.png)
![image](https://user-images.githubusercontent.com/103428684/168936642-386b5583-984c-4af2-8424-70ee60389c2b.png)

**Reply by inytss** [reply]
Dear @olgadiella ,
Untuk package tidyverse, boleh diganti dengan package dplyr atau package yang memang diperlukan dalam tidyverse tersebut dipanggil satu persatu. Untuk plotlynya, boleh dicoba untuk memanggil library plotly pada file ui.R dan server.R nya juga. Selain itu, boleh tahu struktur file yang Ibu gunakan dalam pembuatan dashboard ini serta file apa saja yang sudah disertakan dalam proses deploy?

**Comment by olgadiella** [comment]
Dear mb @inytss 

Berikut saya ss bagian awal dr server, ui, dan global saya ya, serta str dari datanya

server
![image](https://user-images.githubusercontent.com/103428684/169253391-3e4233fc-4dc4-4241-a093-8a0dc12fed3a.png)


ui
![image](https://user-images.githubusercontent.com/103428684/169253420-3d38a4a6-765a-4ed7-ad08-318d6d97e925.png)


global
![image](https://user-images.githubusercontent.com/103428684/169253467-0c42976b-7066-4a54-b307-d6e281a3f52d.png)


str
![image](https://user-images.githubusercontent.com/103428684/169253537-6054d31f-fea3-45d1-bc43-9434ab47b4fc.png)


file yg disertakan saat deploy ke shiny.io:
![image](https://user-images.githubusercontent.com/103428684/169254191-915684cc-c7ff-4224-bc8a-92b1b115d121.png)


sdh coba deploy sesuai usulan dr mb utk tambah library terkait di server dan ui, namun msh failed in deployment, berikut error log terakhir:
![image](https://user-images.githubusercontent.com/103428684/169253961-ae2ce4c4-c941-4025-9d56-36e3721da963.png)
![image](https://user-images.githubusercontent.com/103428684/169253998-bb1fe48e-c7de-4ee8-b059-a3301f8ebb46.png)

**Reply by inytss** [reply]
Dear @olgadiella, 
Apabila saya coba perhatikan saat ini errornya terdapat pada lubridate, yang mana lubridate dipanghil pada file global.R, mungkin Ibu bisa coba untuk memanggil semua library yang dibutuhkan pada file ui.R dan server.R. Selain itu dalam melakukan deploy, file .RData tidak perlu diikutsertakan. Cukup data, global.R, ui.R, serta server.R nya saja.

**Reply by olgadiella** [reply]
Dear mb @inytss 

Saya coba ikuti spt petunjuk mb di atas tp msh blm bs jg :') knp ya mb?

server
![image](https://user-images.githubusercontent.com/103428684/169279666-3f8d1c27-e4fb-470a-b5e4-126de4d99b0f.png)


ui
![image](https://user-images.githubusercontent.com/103428684/169279706-270db1dd-f62e-44ff-9e5e-fd2ca0c23c80.png)


global
![image](https://user-images.githubusercontent.com/103428684/169279746-ba171cb5-aa16-4142-aae9-6e5f5c320e2f.png)


upload
![image](https://user-images.githubusercontent.com/103428684/169279851-3dde4023-9338-4807-96a3-31b0a7921db0.png)


hasil
![image](https://user-images.githubusercontent.com/103428684/169280211-d7c5c0d1-fcf1-4f81-a77a-3e07851d9cd6.png)


log
![image](https://user-images.githubusercontent.com/103428684/169280269-7b6f4ac1-bedf-4546-962e-8f3bbd02116c.png)
![image](https://user-images.githubusercontent.com/103428684/169280314-f5bf7e63-7c77-42db-9584-7390ddfcbf3e.png)

[logs (2).txt](https://github.com/teamalgoritma/community/files/8728548/logs.2.txt)

**Reply by inytss** [reply]
Berdasarkan yang kami coba cek informasi log nya, Ibu boleh coba untuk uninstall kembali package plotly dan install ulang, serta sepertinya pada datanya terdapat karakter yang tidak dapat diconfert dalam format UTF-8. Ibu bisa coba read csv dengan cara berikut
```
read.csv(filepath, encoding="UTF-8")
```

Semoga dapat membantu ya Bu.

**Reply by olgadiella** [reply]
Dear mb @inytss,

Saya sudah coba unpackage plotly dan install ulang, sudah tambahkan encoding juga, namun masih failed juga :'(
Kenapa ya mb? Sudah sampai buat baru lagi dr awal ui, server, global tp msh blm bs jg

Berikut log terakhir:
[logs (3).txt](https://github.com/teamalgoritma/community/files/8747275/logs.3.txt)

![image](https://user-images.githubusercontent.com/103428684/169643489-7df92242-4319-46c9-ac40-5ea6be801974.png)
![image](https://user-images.githubusercontent.com/103428684/169643496-91b51f46-974f-45e5-90bf-175493e82fcb.png)

**Reply by inytss** [reply]
Dear @olgadiella ,
Saya sudah coba cek untuk assignment capstone yang Ibu gunakan dan saya menemukan yang menyebabkan app tidak dapat berjalan yaitu pada code global.R line 19 dalam rename variable `title`. Selanjutnya saya coba untuk deploy app tersebut dan berhasil. Mungkin Ibu bisa mencoba untuk menghapus code pada line 19 untuk rename kolom `title` dan mencoba untuk deploy ulang kembali.

**Reply by inytss** [reply]
Sebagai tambahan saran Ibu bisa mencoba untuk deploy dengan alamat url app yang baru, atau melakukan deploy ulang bukan melakukan re-deploy app.

**Reply by olgadiella** [reply]
Wah ya ampun mb @inytss bener akhirnya bisa up to shiny io!! :') 
Tapi kalo rename dihilangkan, nama column title saat di cek di str di rmd jadi: X.U.FEFF.title
Kalo saya deploy di local dan codingnya hanya pakai "title", saat di run app di local malah ga kebaca, hrs diganti dg nama yg sama spt di atas baru bs diproses.
Tapi kalo saya pake "X.U.FEFF.title" saat deploy ke shiny.io malah jd ada grafik yg ga terbaca.

Akhirnya td trial coba di server pakai nama column "title" biasa aja, meskipun di coba di local ga bisa (krn local bacanya "X.U.FEFF.title"), tapi pas di-deploy ke shiny.io malah berhasil! Bingung knp bs begitu, mgkn krn sdh dimasukin encoding UTF-8 ya saat manggil file

Anyway happy bgt akhirnya solved juga, meskipun sedihnya due saya over a week hehe 
Thankyou ya mb utk supportnya dan kesabarannya memberi feedback dan solving issue saya 👍 

![image](https://user-images.githubusercontent.com/103428684/169822711-5481fc67-928c-4219-b68d-b1903dca0361.png)

https://olgadiella.shinyapps.io/capt_spot/

**Reply by inytss** [reply]
Sepertinya memang dikarenakan adanya encoding pada nama kolom `title` tersebut. Kebetulan jika saya coba dilocal saya, sudah terbaca nama kolomnya `title` bukan `X.U.FEFF.title`. Curiga saya antara pengaturan `encoding="UTF-8"` dan parameter `stringAsFactors=T`. 
Tapi apabila sudah bisa dideploy, saya ikut senang bu :)

Apabila solusi terakhir membantu, Ibu bisa tekan tombol "_Mark as Answered_" supaya tertandai bahwa diskusi ini sudah terjawab :) 👍🏻 
Terima kasih bu :)

**Reply by aitanarP-Heryuda** [reply]
aku sumpah minder lihat capstone nya mba Olga..keren euy!

**Reply by olgadiella** [reply]
huehehe makasih pak @aitanarP-Heryuda, jgn minder sama saya lha pak, saya justru yg msh minder bgt ini msh blm tau banyak, bener2 newbie di R, training ini experienced mantep bgt dah buat saya belajar skill baru! 😄 
Thank you, Team Algoritma! 🌟

**Comment by olgadiella** [comment, marked as answer]
Solved! 
https://github.com/teamalgoritma/community/discussions/112#discussioncomment-2803451

Thanks mb @inytss 👍
